packages/components/src/timeline/index.js

```diff
@@ -0,0 +1,86 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+import moment from 'moment';
+import PropTypes from 'prop-types';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import TimelineGroup from './timeline-group';
+
+const Timeline = ( { className, items } ) => {
+	const timelineClassName = classnames( 'woocommerce-timeline', className );
+
+	if ( ! items || items.length === 0 ) {
+		return (
+			<div className={ timelineClassName }>
+				<p className="timeline_no_events">
+					{ __( 'No events to display', 'woocommerce-admin' ) }
+				</p>
+			</div>
+		);
+	}
+
+	// Sort all the items reverse chronologically
+	items.sort( ( a, b ) => {
+		return a.datetime - b.datetime;
+	} );
+
+	// Group items by days.
+	// TODO: Maybe the items passed into the Timeline component
+	// should already be grouped and sorted?
+	const groups = items.reduce( ( acc, curr ) => {
+		const itemLocalDatetime = moment.unix( curr.datetime );
+		const formattedDate = itemLocalDatetime.format( 'YYYYMMDD' );
+
+		return {
+			...acc,
+			[ formattedDate ]: [ ...( acc[ formattedDate ] || [] ), curr ],
+		};
+	}, {} );
```

Comment:
I think it makes more sense that the items are passed to the timeline grouped and sorted, but I've implemented it like this for now.

An alternative would be to pass something like a flag to the component to indicate whether to group+sort by hour/days/weeks/months/years/etc, but that seems unnecessarily complicated. It's nice for the user, but difficult to implement properly in the component.
The user will arguably have more freedom if they get to group+sort the data however they want and the component doesn't mess with the data.

@timmyc @psealock @allendav I'd love to hear your thoughts on what you think makes the most sense here.

Reply:
What about having the group by functionality be triggered by a prop? Something like `groupBy` that defaults to "day" maybe? And likewise have a prop for sorting order ASC|DESC Then the data could be passed in ungrouped/sorted and the component would take care of the rest. This would also allow for future controls to group items by week, month, quarter etc...

Reply:
Sounds good, I'll do that then 🙂

Reply:
Done in https://github.com/woocommerce/woocommerce-admin/pull/3614/commits/17a835c62c97cca053e0c99cac3f39577a72a8a8 👍